pymc3/model_graph.py

```diff
@@ -40,24 +47,38 @@ def _ancestors(self, var, func, blockers=None):
         return set([j for j in ancestors([func], blockers=blockers) if j in self.var_list and j != var])
 
     def _get_ancestors(self, var, func):
-        """Get all ancestors of a function, doing some accounting for deterministics
-
-        Specifically, if a deterministic is an input, theano.gof.graph.ancestors will
-        return only the inputs *to the deterministic*.  However, if we pass in the
-        deterministic as a blocker, it will skip those nodes.
+        """Get all ancestors of a function, doing some accounting for deterministics.
         """
-        deterministics = self.get_deterministics(var)
+
+        # this contains all of the variables in the model EXCEPT var...
+        vars: List[var] = set(self.var_list)
+        vars.remove(var)
+        
         upstream = self._ancestors(var, func)
```

Comment:
I would completely remove `self._ancestors` and replace it with the `stack_search` that you did